performance-timeline/performanceobservers.js

```diff
@@ -19,3 +19,41 @@ function checkEntries(perfEntriesToCheck, expectedEntries) {
     assert_not_equals(findMatch(pe1), null, "Entry matches");
   });
 }
+
+// Compares a performance entry to a predefined one.
+// perfEntriesToCheck is an array of performance entries from the user agent
+// expectedEntries is an array of performance entries minted by the test
+function checkEntriesInOrder(perfEntriesToCheck, expectedEntries) {
+  assert_equals(perfEntriesToCheck.length, expectedEntries.length, "performance entries must match");
+  for (var i = 0; i < expectedEntries.length; ++i) {
+    var pe = perfEntriesToCheck[i];
+    var ex = expectedEntries[i];
+    if (!pe)
+      continue;
+    assert_equals(ex.name, pe.name, "entry name must match");
+    assert_equals(ex.entryType, pe.entryType, "entry entryType must match");
```

Comment:
I'm not sure what you mean. This method (checkEntriesInOrder) asserts name/type of each entry. The other method (checkEntries) only asserts entries exist with given names/types. Those are similar but different operations.